livox_ros_driver/livox_ros_driver/lddc.cpp

```diff
@@ -202,8 +203,13 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
       }
     }
     /** Use the first packet timestamp as pointcloud2 msg timestamp */
-    if (!published_packet) {
-      cloud.header.stamp = ros::Time(timestamp / 1000000000.0);
+    if (published_packet) {
+      first_timestamp = ros::Time::now();

```

Comment:
How often do you update `first_timestamp`?
If the original. value of `first_timestamp` is not updated, I guess you will experience a large time gap.

Reply:
`first_timestamp` is updated when the really first data arrived and never updated.
original implementation calculated the duration between the first data and the current data.
So I keep the really first timestamp and add the duration to it.